Shopify/cli
# Bundle multiple extension point files

### WHY are these changes introduced?

Fixes [#1899](https://github.com/Shopify/ui-extensions-private/issues/1899)

TLDR; Previously extensions had one `src/index.*` file.  Now there are multiple `src/*.*` files.  Here we update the dev command to build multiple `src/*.*` files into one `dist/main.js` file.

The commit messages go into a lot of detail on the approach I've taken, so please read those.

### WHAT is this pull request doing?

To summarise:

1. Update the dev command to bundle multiple `src/*.*` files into one `dist/main.js` file.
2. Output a message when we build an extension.
3. Update how the app loads the paths to the src files.

But! The commit messages have a lot of detail, so please read those.

### How to test your changes?

Testing Dev:

1. `cd fixtures/app` 
2. `yarn generate extension` (Select the new **Checkout UI Beta** extension under **Shopify private**)
3. `yarn dev`
4. Make a valid change to `/extensions/[name]/src/AdminCheckoutEditor.*`.
  - See a message in the console 
  - Check `/extensions/[name]/dist/` to see your changes
5. Make a valid change to `/extensions/[name]/src/CheckoutDynamicRender.*`.
  - See a message in the console 
  - Check `/extensions/[name]/dist/` to see your changes
6. Make an invalid change to `/extensions/[name]/src/*.*`.
  - See a message in the console 
7. Make a change to an existing extension
  - Check that your changes are built
  - Open the preview link for an existing extension point.  Check that it renders.

Testing Build:

1. `cd fixtures/app` 
2. `yarn generate extension` (Select the new **Checkout UI Beta** extension under **Shopify private**)
3. `yarn build`

The new extension should be built: 

<img width="460" alt="Screenshot 2022-11-16 at 10 36 37 AM" src="https://user-images.githubusercontent.com/690791/202224669-4875e0e5-70e0-4122-a796-44499c8434c8.png">

Testing validation with `yarn dev` && `yarn build`:

1. `cd fixtures/app` 
2. `yarn generate extension` (Select the new **Checkout UI Beta** extension under **Shopify private**)
3. Add a new extension point for the newly created toml file whose src file does not exist.  Run `yarn dev` or `yarn build`, you should see an error
4. In the toml file duplicate an extension point that already exists.  Run `yarn dev` or `yarn build`, you should see a different message.
5. Try different combinations of errors (module path is incorrect & duplicate extension points).  The error message should always make sense

<img width="568" alt="Screenshot 2022-11-16 at 10 32 33 AM" src="https://user-images.githubusercontent.com/690791/202226714-1868f0fe-bedf-408f-b798-af305a6e7965.png">

**Note this isn't working right now, I'm investigating**

1. `cd fixtures/app` 
2. `yarn generate extension` (Select the new **Checkout UI Beta** extension under **Shopify private**)
3. `yarn shopify app info`.  You should see error messages for the newly created extension





### Post-release steps
None

### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
